pywinauto/application.py

```diff
@@ -583,14 +583,15 @@ def print_control_identifiers(self, depth=None, filename=None):
         for name, control in name_control_map.items():
             control_name_map.setdefault(control, []).append(name)
 
+        log_func = print
         log_file = None
-        if filename is None:
-            print("Control Identifiers:")
-        else:
+        if filename is not None:
             log_file = open(filename, "w")
-            log_file.write("Control Identifiers:")
+            log_func = log_file.write

```

Comment:
Sorry, forgot to say. It will produce static warning since it's a named function so using `def log_func` is preferable in this condition branch.